gptel-transient.el

```diff
@@ -449,12 +473,33 @@ Customize `gptel-directives' for task-specific prompts."
     (gptel--suffix-rewrite-and-ediff)]]
   (interactive)
   (unless gptel--rewrite-message
-    (setq gptel--rewrite-message (gptel--rewrite-message)))
+    (setq gptel--rewrite-message (funcall gptel-rewrite-message)))
   (transient-setup 'gptel-rewrite-menu))
 
-
 ;; * Transient Infixes
 
+;; ** Infixes for context aggregation
+
+(transient-define-infix gptel--infix-context-destination ()
```

Comment:
With more than 2 options, I think using a transient switch doesn't make sense, since you can't see all the options on screen. Pressing `-xd` is a "blind" operation.  It's better for this to be a simple `lisp-variable` transient class.

Reply:
I haven't had an existing similar example so I winged it. Should the variables be as-is (keywords) or should they have a string representation? I figured a string representation would be more intuitive. As far as am aware, unless you have Helm mode enabled, you won't see an autocomplete option, and thus won't know what options to write in.

Reply:
By default, pressing `TAB` will show all possible completions in completing-read and complete it for you.  With any popular completion system (helm/ivy/vertico/fido/icomplete), the options will be available in a list.

`completing-read` can complete symbols, the candidates don't need to be strings.

Reply:
Examples of the `lisp-variable` transient class are all over `gptel-transient`.  See for example the infix for setting the model+backend.

Reply:
Done. I made the settings respect buffer locality as well, but it doesn't work for the switches due to #326.